docs/guide/batch-operations.md

```diff
@@ -0,0 +1,222 @@
+
+```javascript
+messages: {
+  progress: 'Resetting {{ type }}...',
+  completed: 'Reset {{ count }} {{ type }}.'
+},
+```
+
+The `messages` object properties are used in notifications that appear to tell the editor what is happening behind the scenes. The `progress` message appears when the operation begins and the `completed` messages appears when it is done.
+
+They both can use the `type` interpolation key, which Apostrophe replaces with the piece type label. The `completed` message can also include a `count` interpolation key, which is replaced by the number of pieces that were updated.
+
+```javascript
+if: {
+  archived: false
+},
+```
+
+`if` is an optional property that allows you to define filter conditions when the option is available. In this case, the "Reset" operation is only available when the `archived` filter is `false` (the editor is not looking at archived pieces). This might be because archived pieces should be left as they are and not reset to their defaults. This property works similar to [conditional schema fields](/guide/conditional-fields.md), but in this case the conditions are for manager filters, not fields.
+
+```javascript
+modalOptions: {
+  title: 'Reset {{ type }}',
+  description: 'Are you sure you want to reset {{ count }} {{ type }}?',
+  confirmationButton: 'Yes, reset the selected content'
+}
+```
+
+The `modalOptions` object configures the confirmation modal that appears when an editor initiates a batch operation. This confirmation step helps to prevent accidental changes to possibly hundreds of pieces. If this is not included, the batch operation's `label` is used for the title, there is no description, and the standard confirmation button label is used (e.g., "Yes, continue.").
+
+With these configuration, we should immediately see a button for the "Reset" operation in the article piece manager.
+
+![The article piece manager, now with a button using the recycle symbol](/images/batch-operation-recycle-button.png)
+
+## Adding the API route
+
+Right now if we clicked that new button and confirmed to continue nothing would happen except for an error notification saying something like "Batch operation Reset failed." Since the batch operation is called `reset`, the manager is going to look for an API route at `/v1/api/article/reset` (the piece type's base API path, plus `/reset`). We need to add that route to the piece type.
+
+Batch operation route handlers will usually have a few steps in common, so we can look at those elements in the example below.
+
+<AposCodeBlock>
+  ```javascript
+  module.export = {
+    // `batchOperations` and other module settings...
+    apiRoutes(self) {
+      return {
+        post: {
+          reset(req) {
+            // Make sure there is an `_ids` array provided.
+            if (!Array.isArray(req.body._ids)) {
```

Comment:
Use `self.apos.launder.ids` to make sure you wind up with strings in that array, avoiding crashes or extra sanitization steps for malicious input

(It's OK to still check if it's an array first if you want and throw invalid if it's not)

Reply:
Okay, we don't do that in our archive and restore routes. Is it not necessary or should we add it there? I recall you saying that since we have this coming from the Apos UI that extra launder wasn't necessary.

Reply:
Oof... if I said that I was mistaken. You can never trust the browser.

That said, it's not crazy urgent, it's only an unproven hypothetical that something other than an id string could cause a problem here and it certainly can't cause a security breach. I will open a ticket to follow up implementation-wise, if you'd please take care of it here in the docs example.